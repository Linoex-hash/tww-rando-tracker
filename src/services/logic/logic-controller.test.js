import LogicController from './logic-controller';

describe('LogicController', () => {
  describe('allItems', () => {
    test('returns a list of all the items, including entrances, charts, and keys', () => {
      const allItems = LogicController.allItems();

      expect(allItems).toMatchSnapshot();
    });
  });

  describe('shortDungeonName', () => {
    test('returns the short dungeon name based on the dungeon name', () => {
      const shortDungeonName = LogicController.shortDungeonName('Dragon Roost Cavern');

      expect(shortDungeonName).toEqual('DRC');
    });
  });

  describe('dungeonEntryName', () => {
    test('returns the entry name based on a dungeon name', () => {
      const entryName = LogicController.dungeonEntryName('Dragon Roost Cavern');

      expect(entryName).toEqual('Entered DRC');
    });
  });

  describe('shortCaveName', () => {
    test('returns the cave name without mentioning secret caves', () => {
      const shortCaveName = LogicController.shortCaveName('Dragon Roost Island Secret Cave');

      expect(shortCaveName).toEqual('Dragon Roost Island Cave');
    });

    test('returns the cave name without mentioning warp maze caves', () => {
      const shortCaveName = LogicController.shortCaveName('Diamond Steppe Island Warp Maze Cave');

      expect(shortCaveName).toEqual('Diamond Steppe Island Cave');
    });
  });

  describe('caveEntryName', () => {
    test('returns the entry name based on a cave name', () => {
      const entryName = LogicController.caveEntryName('Dragon Roost Island Secret Cave');

      expect(entryName).toEqual('Entered Dragon Roost Island Cave');
    });
  });
});
