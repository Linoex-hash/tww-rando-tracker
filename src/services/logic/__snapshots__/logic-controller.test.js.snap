// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogicController allItems returns a list of all the items, including entrances, charts, and keys 1`] = `
Array [
  "Entered Savage Labyrinth",
  "Entered Dragon Roost Island Cave",
  "Entered Fire Mountain Cave",
  "Entered Ice Ring Isle Cave",
  "Entered Cabana Labyrinth",
  "Entered Needle Rock Isle Cave",
  "Entered Angular Isles Cave",
  "Entered Boating Course Cave",
  "Entered Stone Watcher Island Cave",
  "Entered Overlook Island Cave",
  "Entered Bird's Peak Rock Cave",
  "Entered Pawprint Isle Chuchu Cave",
  "Entered Pawprint Isle Wizzrobe Cave",
  "Entered Diamond Steppe Island Cave",
  "Entered Bomb Island Cave",
  "Entered Rock Spire Isle Cave",
  "Entered Shark Island Cave",
  "Entered Cliff Plateau Isles Cave",
  "Entered Horseshoe Island Cave",
  "Entered Star Island Cave",
  "Treasure Chart 25",
  "Treasure Chart 7",
  "Treasure Chart 24",
  "Triforce Chart 2",
  "Treasure Chart 11",
  "Triforce Chart 7",
  "Treasure Chart 13",
  "Treasure Chart 41",
  "Treasure Chart 29",
  "Treasure Chart 22",
  "Treasure Chart 18",
  "Treasure Chart 30",
  "Treasure Chart 39",
  "Treasure Chart 19",
  "Treasure Chart 8",
  "Treasure Chart 2",
  "Treasure Chart 10",
  "Treasure Chart 26",
  "Treasure Chart 3",
  "Treasure Chart 37",
  "Treasure Chart 27",
  "Treasure Chart 38",
  "Triforce Chart 1",
  "Treasure Chart 21",
  "Treasure Chart 6",
  "Treasure Chart 14",
  "Treasure Chart 34",
  "Treasure Chart 5",
  "Treasure Chart 28",
  "Treasure Chart 35",
  "Triforce Chart 3",
  "Triforce Chart 6",
  "Treasure Chart 1",
  "Treasure Chart 20",
  "Treasure Chart 36",
  "Treasure Chart 23",
  "Treasure Chart 12",
  "Treasure Chart 16",
  "Treasure Chart 4",
  "Treasure Chart 17",
  "Treasure Chart 31",
  "Triforce Chart 5",
  "Treasure Chart 9",
  "Triforce Chart 4",
  "Treasure Chart 40",
  "Triforce Chart 8",
  "Treasure Chart 15",
  "Treasure Chart 32",
  "Treasure Chart 33",
  "Entered DRC",
  "Entered FW",
  "Entered TotG",
  "Entered FF",
  "Entered ET",
  "Entered WT",
  "Entered GT",
  "Tingle Tuner",
  "Wind Waker",
  "Spoils Bag",
  "Grappling Hook",
  "Power Bracelets",
  "Iron Boots",
  "Bait Bag",
  "Boomerang",
  "Hookshot",
  "Delivery Bag",
  "Bombs",
  "Skull Hammer",
  "Deku Leaf",
  "Hero's Shield",
  "Mirror Shield",
  "Triforce Shard",
  "Nayru's Pearl",
  "Din's Pearl",
  "Farore's Pearl",
  "Wind's Requiem",
  "Ballad of Gales",
  "Command Melody",
  "Earth God's Lyric",
  "Wind God's Aria",
  "Song of Passing",
  "Boat's Sail",
  "Note to Mom",
  "Maggie's Letter",
  "Moblin's Letter",
  "Cabana Deed",
  "Tingle Statue",
  "Magic Meter Upgrade",
  "Ghost Ship Chart",
  "Progressive Sword",
  "Progressive Bow",
  "Progressive Wallet",
  "Progressive Picto Box",
  "Empty Bottle",
  "Telescope",
  "Magic Armor",
  "Hero's Charm",
  "Progressive Quiver",
  "Progressive Bomb Bag",
  "Hurricane Spin",
  "DRC Big Key",
  "DRC Small Key",
  "FW Big Key",
  "FW Small Key",
  "TotG Big Key",
  "TotG Small Key",
  "ET Big Key",
  "ET Small Key",
  "WT Big Key",
  "WT Small Key",
]
`;
